docs/content/addons/cosi.md

```diff
@@ -0,0 +1,32 @@
++++
+title = " Container Object Storage Interface (COSI)"
+icon = "fa-solid fa-eye"
++++
+
+By leveraging CAPI cluster lifecycle hooks, this handler deploys [Container Object Storage Interface] (COSI)
+on the new cluster at the `AfterControlPlaneInitialized` phase.
+
+Deployment of COSI is opt-in via the [provider-specific cluster configuration]({{< ref ".." >}}).
+
+The hook uses the [Cluster API Add-on Provider for Helm] to deploy the NFD resources.
```

Comment:
I believe you meant deploying the COSI resources?

Reply:
I was so close to hiding the copy and paste here :)